libraries/src/MVC/Model/AdminModel.php

```diff
@@ -487,6 +494,21 @@ protected function batchCopy($value, $pks, $contexts)
 			// Get the new item ID
 			$newId = $this->table->get('id');
 
+			if (!empty($oldAssetId))
+			{
+				// Copy rules
+				$query = $db->getQuery(true);
+				$query->clear()
+					->update($db->quoteName('#__assets', 't'))
+					->join('INNER', $db->quoteName('#__assets', 's') .
+						' ON ' . $db->quoteName('s.id') . ' = ' . $oldAssetId
+					)
+					->set($db->quoteName('t.rules') . ' = ' . $db->quoteName('s.rules'))
+					->where($db->quoteName('t.id') . ' = ' . $this->table->asset_id);
```

Comment:
$this->table->asset_id should be a prepared parameter
```suggestion
					->where($db->quoteName('t.id') . ' = :oldAssetId2')
					->bind([':oldAssetId1', ':oldAssetId2'], $this->table->asset_id, ParameterType::INTEGER);
```

This requires and use statement for ParameterType and you should remove the if statement at line 497 as george suggested.